Modules/mathmodule.c

```diff
@@ -2016,6 +2016,62 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+    if (a == NULL) {
+        return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        Py_DECREF(a);
+        return NULL;
+    }
+    if (_PyLong_Sign(a) == 0 || _PyLong_Sign(b) == 0) {
+        Py_DECREF(a);
+        Py_DECREF(b);
+        return PyLong_FromLong(0);
+    }
+    else {
```

Comment:
This is partly a matter of personal style, but I'd suggest getting rid of the `else` here. There's no need for it, since the `if` branch returns, and by getting rid of it you pull the main logic up one level; I find that more readable.